ambry-cloud/src/main/java/com/github/ambry/cloud/VcrReplicationManager.java

```diff
@@ -292,4 +338,66 @@ protected String getReplicaThreadName(String datacenterToReplicateFrom, int thre
   protected boolean shouldReplicateFromDc(String datacenterName) {
     return cloudConfig.vcrSourceDatacenters.contains(datacenterName);
   }
+
+  /**
+   * Check and schedule a VCR helix update task. We schedule updateVcrHelix in 2 cases:
+   * 1. On a node become online role of partition 1
+   * 2. On Ambry cluster change
+   */
+  private void scheduleVcrHelix() {
+    if (vcrHelixUpdateFuture != null && vcrHelixUpdateFuture.cancel(false)) {
+      // If a vcrHelixUpdate task is scheduled, try to cancel it first.
+      logger.info("There was a scheduled vcrHelixUpdate task. Canceled.");
+    }
+    // either success cancel or not, we should schedule a new job to updateVcrHelix
+    vcrHelixUpdateFuture =
+        scheduler.schedule(() -> updateVcrHelix(), cloudConfig.vcrHelixUpdateDelayTimeInSeconds, TimeUnit.SECONDS);
+    logger.info("VcrHelixUpdate task scheduled. Will run in {} seconds.", cloudConfig.vcrHelixUpdateDelayTimeInSeconds);
+  }
+
+  /**
+   * The actual performer to update VCR Helix:
+   */
+  private void updateVcrHelix() {
+    String localDcZkStr = ((HelixClusterManager) clusterMap).getLocalDcZkConnectString();
+    logger.info("Going to update VCR Helix Cluster. Dryrun: {}", cloudConfig.vcrHelixUpdateDryRun);
+    logger.info("Current partitions in clustermap: ", clusterMap.getAllPartitionIds(null));
```

Comment:
Theoretically its possible for cluster map to get updated between line 363 and 364. It might be better to `getAll{artitoinIds` to a local list, and use the same list to log as well as for updating.

Reply:
Added some log in `updateResourceAndPartition` to give us more precise information.